### DISCUSSION TITLE
Marker-Type?

### DISCUSSION THREAD

**Discussion by ramGHnet** [javalent/obsidian-leaflet#41] [Q&A]
Hi,

I read the docs three times, but I couldn't find out how I can use the <type> in
marker: <type>,....

In my case when I add a marker by right-click on the map the marker is shown as gray icon
and is very bad to see. 
Also, when I add a marker row to the code block the marker is shown as gray icon and are very
bad to see.

How can I change this?

Best regards
Marten

**Comment by valentine195** [comment]
Hi,

Additional marker "types" can be created in the settings page of the plugin. When you create a new marker type, you can customize the color, icon, and icon placement (if you have it layered on the base marker). Once additional types have been created, right clicking on the map will have these as options. 

Creating a marker in the map block directly should look like this:

`marker: default,32,-87,[[Note]]`

This will create a marker at latitude 32, longitude -87 using the "default" marker type. The marker will also be linked to Note.md.

**Reply by valentine195** [reply]
Also, let's say in the settings you create a marker type named "person". You can do this:

`marker: person, 32, -87, [[Note]]`

**Comment by ramGHnet** [comment]
Ah ok. I use Leaflet mostly in Jupyter Notebooks (Python). There you can create different markers in every notebook.
In Obsidian-Leaflet I have to create them in the settings for all notes.

**Reply by valentine195** [reply]
Is that using [this plugin](https://ipyleaflet.readthedocs.io/en/latest/)?

Seems like it executes python inline...probably beyond the scope of this plugin 😄 

Thanks for bringing it to my attention, though. I'll take a look through the docs for any improvements I could make. Are there any killer features that plugin has that you'd like to see? (no guarantees they'll make it in, though)

**Reply by ramGHnet** [reply]
We use ipyleaflet for big data, for example millions of real estate ads. There heatmaps, marker clusters etc. are interesting.
But in obsidian-leaflet rather not. One feature could be that you can include more information in the marker popus. For example, a preview of the associated note or similar.

For obsidian-leaflet I would wish a clearer structure and above all better documentation.

I've been trying to figure out how to create custom markers in obsidian-leaflet for what seems like an eternity. Where to put the images (icons) and so on. Here would be an example and better documentation helpful

**Reply by valentine195** [reply]
Thanks for the feedback.

Please see [Additional Markers](https://github.com/valentine195/obsidian-leaflet-plugin#additional-markers) for a reference on how to create additional markers. I did just improve this documentation somewhat, but this section is relatively unchanged.

For a step by step process:

1. Open the plugin "settings" tab.
2. Scroll to Additional Markers
3. Click the "Add Additional" button
4. The following modal will pop up:
![image](https://user-images.githubusercontent.com/38669521/117870564-42af6300-b26a-11eb-9c66-c72dd3fcab2e.png)
5. Enter the following:
      -  Marker Name (must be unique - this is what you would use in the `marker:` tag or what is available in the `Marker Type` drop down when right-clicking)
      - Marker Icon (this is the _icon name_ from [Font Awesome Free](https://fontawesome.com/icons?d=gallery&p=2&s=solid&m=free). In 3.8.1, I improved the experience of choosing an icon here.
      - Optional: Layer Marker - Layer Marker on top of the "default marker"
      - Optional: Marker Color - override default color
6. Optional: If "layer marker" is turned on, you can edit _where_ the marker is positioned by clicking and dragging the icon. Holding <kbd>Shift</kbd> will snap the marker to the midlines.
7. Click save

This marker type will now be available in the right-click menu on the map.

**Reply by valentine195** [reply]
Did this help you?
